packages/firestore/src/core/target_id_generator.ts

```diff
@@ -31,44 +32,54 @@ enum GeneratorIds {
  * may group IDs from separate parts of the client into the same ID space.
  */
 export class TargetIdGenerator {
```

Comment:
At this point this is a fancy class to add 2 to numbers.  next() adds 2 to the last value returned by after() or next(), and after() adds 2 to the value you pass in.  Perhaps rename it to AddTwoer for clarity?

Or maybe we should drop it and just define a few constants (either in types.ts next to the TargetId definition or perhaps we should create a target_id.ts file or something).

```
const FIRST_LIMBO_TARGET_ID = 1;
const FIRST_LOCAL_STORE_TARGET_ID = 2;
const TARGET_ID_INCREMENT = 2;
```
(with some appropriate comments)

I don't care super strongly but I do think the TargetIdGenerator abstraction isn't paying for itself.  And it's a bit suspicious that it has next() and after() methods but 1 consumer exclusively uses next() [limbo resolution] and the other exclusively uses after() [user queries]...  This also means that after() could literally just add 2 without updating this.nextID.

As-is I almost feel like the comments should call out that this is just a "add 2" helper. :-)

Reply:
This is a fair comment, but after some extensive googling, I wasn't able to find an NPM package we could substitute for this generator.

... but in reality, I'd like to keep it as is. It more closely matches what we are doing on other platforms. I am not opposed to removing eventually for all clients (TODO added), but would prefer to defer this work until later.